Brokerages/Alpaca/AlpacaBrokerage.cs

```diff
@@ -0,0 +1,415 @@
+
+            _sockClient.ConnectAsync().SynchronouslyAwaitTask();
+            _natsClient.Open();
+
+            _isConnected = true;
+
+            // create new thread to manage disconnections and reconnections
+            _cancellationTokenSource = new CancellationTokenSource();
+            _connectionMonitorThread = new Thread(() =>
+            {
+                var nextReconnectionAttemptSeconds = 1;
+
+                try
+                {
+                    while (!_cancellationTokenSource.IsCancellationRequested)
+                    {
+                        var isAlive = true;
+                        try
+                        {
+                            isAlive = _sockClient.IsAlive;
+                        }
+                        catch (Exception)
+                        {
+                            // ignored
+                        }
+
+                        if (isAlive && _connectionLost)
+                        {
+                            _connectionLost = false;
+                            nextReconnectionAttemptSeconds = 1;
+
+                            OnMessage(BrokerageMessageEvent.Reconnected("Connection with Alpaca server restored."));
+                        }
+                        else if (!isAlive)
+                        {
+                            if (_connectionLost)
+                            {
+                                try
+                                {
+                                    Thread.Sleep(TimeSpan.FromSeconds(nextReconnectionAttemptSeconds));
+
+                                    _sockClient.ConnectAsync().SynchronouslyAwaitTask();
+                                }
+                                catch (Exception exception)
+                                {
+                                    // double the interval between attempts (capped to 1 minute)
+                                    nextReconnectionAttemptSeconds = Math.Min(nextReconnectionAttemptSeconds * 2, 60);
+
+                                    Log.Error(exception);
+                                }
+                            }
+                            else
+                            {
+                                _connectionLost = true;
+
+                                OnMessage(
+                                    BrokerageMessageEvent.Disconnected(
+                                        "Connection with Alpaca server lost. " +
+                                        "This could be because of internet connectivity issues. "));
+                            }
+                        }
+
+                        Thread.Sleep(1000);
+                    }
+                }
+                catch (Exception exception)
+                {
+                    Log.Error(exception);
+                }
+            })
+            { IsBackground = true };
+            _connectionMonitorThread.Start();
+            while (!_connectionMonitorThread.IsAlive)
```

Comment:
I think there is a small risk of staying trapped in this `while` loop if the `_connectionMonitorThread` stops in between checks. Me just being paranoid.